lib/ansible/modules/database/postgresql/postgresql_facts.py

```diff
@@ -0,0 +1,916 @@
+                              "WHERE table_name = 'pg_replication_slots')")
+        if not res[0][0]:
+            return True
+
+        query = ("SELECT slot_name, plugin, slot_type, database, "
+                 "active FROM pg_replication_slots")
+        res = self.__exec_sql(query)
+
+        # If there is no replication:
+        if not res:
+            return True
+
+        rslot_dict = {}
+        for i in res:
+            rslot_name = i[0]
+            rslot_info = {}
+            rslot_info["plugin"] = i[1]
+            rslot_info["slot_type"] = i[2]
+            rslot_info["database"] = i[3]
+            rslot_info["active"] = i[4]
+
+            rslot_dict[rslot_name] = rslot_info
+
+        self.pg_facts["repl_slots"] = rslot_dict
+
+    def get_settings(self):
+        """
+        Get server settings.
+        """
+        query = ("SELECT name, setting, unit, context, vartype, "
+                 "boot_val, min_val, max_val, sourcefile "
+                 "FROM pg_settings")
+        res = self.__exec_sql(query)
+
+        set_dict = {}
+        for i in res:
+            set_name = i[0]
+            set_info = {}
+            set_info["setting"] = i[1]
+
+            if i[2]:
+                set_info["unit"] = i[2]
+            else:
+                set_info["unit"] = ""
+
+            set_info["context"] = i[3]
+            set_info["vartype"] = i[4]
+            if i[5]:
+                set_info["boot_val"] = i[5]
+            else:
+                set_info["boot_val"] = ""
+
+            if i[6]:
+                set_info["min_val"] = i[6]
+            else:
+                set_info["min_val"] = ""
+
+            if i[7]:
+                set_info["max_val"] = i[7]
+            else:
+                set_info["max_val"] = ""
+
+            if i[8]:
+                set_info["sourcefile"] = i[8]
+            else:
+                set_info["sourcefile"] = ""
+
+            set_dict[set_name] = set_info
+
+        self.pg_facts["settings"] = set_dict
+
+    def get_repl_info(self):
+        """
+        Get information about replication if the server is a master.
+        """
+        # Check that pg_replication_slots exists:
+        res = self.__exec_sql("SELECT EXISTS (SELECT 1 FROM "
+                              "information_schema.tables "
+                              "WHERE table_name = 'pg_stat_replication')")
+        if not res[0][0]:
+            return True
+
+        query = ("SELECT r.pid, a.rolname, r.application_name, r.client_addr, "
+                 "r.client_hostname, r.backend_start::text, r.state "
+                 "FROM pg_stat_replication AS r "
+                 "JOIN pg_authid AS a ON r.usesysid = a.oid")
+        res = self.__exec_sql(query)
+        repl_dict = {}
+
+        # If there is no replication:
+        if not res:
+            return True
+
+        for i in res:
+            repl_pid = i[0]
+            repl_info = {}
+            repl_info["usename"] = i[1]
```

Comment:
Should this not be **username** ?
Why not create the dict with some values in it ?
```suggestion
            repl_info = dict(username=i[1])
```